### ISSUE TITLE
Combine latest on some

### ISSUE THREAD

**Issue by CJ-Wright** [mrocklin/streams#22]
In conversation with @ordirules and @mrocklin 

The idea is that you may only want combine latest to trigger on some of the updates.

**Comment by ordirules** [comment]
I think for simplicity, maybe `combine_latest` should just assume that you only emit on the first child, which means changing this line in `combine_latest`:
```python
        if not self.missing and who == self.children[0]:
```

thus this will emit once:
```python
s1 = Stream()
s2 = Stream()
sout = s1.combine_latest(s2)
sout.map(print)

s2.emit(1)
s1.emit(1)
```

In the long run, I think a trigger should be added. Perhaps add a `trigger` flag to the `buffer` stream element?

What do you think? I think it's simpler than anything else I can think of. Also, we should fork streams itself (separate from event model). It could be on your or my branch, up to you.

**Comment by mrocklin** [comment]
There are definitely cases where people want to emit on updates from
multiple children.  I think that one would take in a list of booleans in a
trigger= or emit_on= keyword for which children should activate an emit.

On Mon, Jun 12, 2017 at 10:52 AM, Julien Lhermitte <notifications@github.com

**Comment by CJ-Wright** [comment]
I like the idea of a kwarg for this.

I guess it would be like:
```python
emiting_children = [child for child, t in zip(self.children, trigger) if t is True]
if not self.missing and who in allowed_emiting:
```

**Comment by ordirules** [comment]
Sorry yes, that was too restrictive. I like that too.
Maybe have the keyword be the list (or one element) of numerically ordered children as a compromise?
(so `s.combine_latest(emit_on=0)` would be fine)
I suggest a PR somewhere? Maybe @CJ-Wright self-PR from your feature branch to your `master` and we can work from there? (Unless @mrocklin  you want to do it)

**Comment by CJ-Wright** [comment]
Closed by #32
